src/scenes/contact/contact.tsx

```diff
@@ -1,5 +1,34 @@
-import React from "react";
+import React, { useEffect, useState, useRef } from "react";
+import { useNavContext } from "../../data/navProvider";
 import styles from "./contact.module.scss";
+
 export function Contact() {
-  return <section id="contact" className={styles.container}></section>;
+  const [position, setPosition] = useState(0);
```

Comment:
this resize related logic could be a custom hook, because you use it in several places. But why excacly do you need to calculate this position?

Reply:
Position of each section is compared with scrollYOffset of a window and it sets an active section for menu.

Reply:
usePosition hook is ready. I hope you meant something like this.